### DISCUSSION TITLE
Sampling from a distribution

### DISCUSSION THREAD

**Discussion by verdverm** [getsynth/synth#341] [Q&A] [answered]
I'd like to generate data where X% of users have a value set, perhaps having several (overlapping) options like linked accounts or something.

Is this possible or now a feature request?

**Comment by AlexMikhalev** [comment, marked as answer]
You can adjust the frequency of categorical fields:
```
{
  "type": "string",
  "categorical": {
    "USD": 8,
    "GBP": 1,
    "EUR": 1
  }
}
```
USD will appear eight times more likely than GBP/EUR.
You can sample from the distribution:
```
{
  "type": "number",
  "subtype": "f64",
  "range": {
    "low": 274.4,
    "high": 6597.5,
    "step": 0.01
  }
}
```

or sample from data source:
```
{
  "type": "datasource",
  "path": "json:descriptions.json",
  "cycle": true
}
```
where descriptions

```
[
  "Walmart Credit Purchase",
  "Amazon - Kindle reading tablet",
  "Apple store subscription",
  "Google store rental",
  "Apple accessories",
  "Amazon Kindle subscription"
]
```
see [Tweaking individual fields](https://www.getsynth.com/docs/examples/bank#tweaking-individual-fields)
I am converting this issue to discussion for others' benefit.
